pallets/creator-staking/src/lib.rs

```diff
@@ -0,0 +1,748 @@
+
+                consumed_weight + T::DbWeight::get().reads_writes(5, 3)
+            } else {
+                T::DbWeight::get().reads(4)
+            }
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight(Weight::from_ref_time(100_000) + T::DbWeight::get().reads_writes(3, 1))]
+        pub fn force_register_creator(
+            origin: OriginFor<T>,
+            space_id: SpaceId,
+        ) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            ensure_root(origin)?;
+
+            ensure!(
+                !RegisteredCreators::<T>::contains_key(space_id),
+                Error::<T>::AlreadyUsedCreatorSpace,
+            );
+
+            let space_owner = T::SpacesInterface::get_space_owner(space_id)?;
+            T::Currency::reserve(&space_owner, T::RegistrationDeposit::get())?;
+
+            RegisteredCreators::<T>::insert(space_id, CreatorInfo::new(space_owner.clone()));
+
+            Self::deposit_event(Event::<T>::CreatorRegistered { who: space_owner, space_id });
+
+            Ok(().into())
+        }
+
+        #[pallet::call_index(1)]
+        #[pallet::weight(Weight::from_ref_time(100_000) + T::DbWeight::get().reads_writes(2, 1))]
+        pub fn unregister_creator(
+            origin: OriginFor<T>,
+            space_id: SpaceId,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+
+            Self::do_unregister_creator(space_id, UnregistrationAuthority::Creator(who.clone()))?;
+
+            Self::deposit_event(Event::<T>::CreatorUnregistered { who, space_id });
+
+            Ok(().into())
+        }
+
+        #[pallet::call_index(2)]
+        #[pallet::weight(Weight::from_ref_time(100_000) + T::DbWeight::get().reads_writes(2, 1))]
+        pub fn force_unregister_creator(
+            origin: OriginFor<T>,
+            space_id: SpaceId,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+
+            Self::do_unregister_creator(space_id, UnregistrationAuthority::Root)?;
+            let creator_info =
+                Self::registered_creator(space_id).ok_or(Error::<T>::NotRegisteredCreator)?;
+
+            Self::deposit_event(Event::<T>::CreatorUnregisteredWithSlash {
+                who: creator_info.stakeholder,
+                space_id,
+                amount: T::RegistrationDeposit::get(),
+            });
+
+            Ok(Pays::No.into())
+        }
+
+        #[pallet::call_index(3)]
+        #[pallet::weight(Weight::from_ref_time(10_000))]
+        pub fn stake(
+            origin: OriginFor<T>,
+            space_id: SpaceId,
+            #[pallet::compact] amount: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            let staker = ensure_signed(origin)?;
+
+            // Check that creator is ready for staking.
```

Comment:
```suggestion
            ensure!(amount > Zero::zero(), Error::<T>::CannotStakeZero);

            // Check that creator is ready for staking.
```

Reply:
It's not only about the `amount` specified in parameters. We do calculate the amount, which staker can stake. It may be 0:
- Minimum stake is 100
- `MinimumRemainingFreeBalance` is 100
- You try to stake 100
  - You'll get an error: `CannotStakeZero`